authenticate-with-oauth/README.md

```diff
@@ -12,71 +12,57 @@ Your app may need to access items that are only shared with authorized users. Fo
-
-#### Setting up the Manifest.xml:
-
-1. Set the launch mode for the `OAuthUserSignInActivity` to route it's intent instance through a call to its `onNewIntent()` method, rather than creating a new instance of the activity. To find out more on setting the launch configuration of an activity, visit the [Android docs](https://developer.android.com/guide/topics/manifest/activity-element)
+1. Create an `AuthenticatorState` and set the `OAuthConfiguration` with the portal URL, client ID, and redirect URI.
+2. Load a `PortalItem(...)` with connection type `Portal.Connection.Authenticated` which will issue an authentication challenge.
+3. Create a `DialogAuthenticator` composable passing it the `AuthenticatorState` object. Because the `AuthenticatorState` object has an `OAuthConfiguration` set, the `DialogAuthenticator` will prompt for OAuth credentials.
+4. Set up the manifest.xml to handle the OAuth redirect URI.
+   * Define a second activity in the `manifest.xml` with the a name `OAuthUserSignInActivity` from the toolkit
 
    ```xml
-   <activity 
-             android:name=".OAuthUserSignInActivity"
-             android:launchMode="singleTop"
-             ...
+   <activity
+        android:name="com.arcgismaps.toolkit.authentication.OAuthUserSignInActivity"
+        android:launchMode="singleTop"> <!--keeps only one instance to the top of the stack-->
    ```
 
-2. Set the `<intent-filter>` categories tags to be able to launch a custom browser tab.
+   * Set the `<intent-filter>` categories tags to be able to launch a custom browser tab.
 
    ```xml
-   <category android:name="android.intent.category.DEFAULT" />
-   <category android:name="android.intent.category.BROWSABLE" />
+   <intent-filter>
+        <action android:name="android.intent.action.VIEW" />
+        <!--required to launch a custom browser tab-->
+        <category android:name="android.intent.category.DEFAULT" />
+        <category android:name="android.intent.category.BROWSABLE" />
    ```
 
-3. Set the `data` tag to be able to use the redirect URI to navigate back to the app after prompting for OAuth credentials. To learn more on setting up the data specification to an intent filter, visit the [Android docs](https://developer.android.com/guide/topics/manifest/data-element).
+   * Set the `data` tag to be able to use the redirect URI to navigate back to the app after prompting for OAuth credentials.
 
    ```xml
    <data
-       android:host="auth"
-       android:scheme="my-ags-app" />
+        android:host="auth"
+        android:scheme="my-ags-app" />
    ```
 
+To learn more on setting up the data specification to an intent filter, visit the [Android docs](https://developer.android.com/guide/topics/manifest/data-element).
+
 ## Relevant API
 
 * ArcGISAuthenticationChallengeHandler
 * ArcGISAuthenticationChallengeResponse
 * AuthenticationManager
-* OAuthUserConfiguration
-* OAuthUserCredential
-* OAuthUserSignIn
+* AuthenticatorState
+* DialogAuthenticator
```

Comment:
Somewhere in this doc, probably more at the top, it should be highlighted that this sample makes use of the toolkit's `authentication` module, possibly with a [link](https://github.com/Esri/arcgis-maps-sdk-kotlin-toolkit/tree/main/toolkit/authentication) to that module in the toolkit repo.